### DISCUSSION TITLE
Need help to add to default whichkey keymap for telescope

### DISCUSSION THREAD

**Discussion by juridiener** [LazyVim/LazyVim#3159] [Q&A]
Hello, I would like to add an additional mapping under telescope to which key. 
Namely, `<leader>fs` should open document symbols. How can I do that? I don't have much experience...
Appreciate any help.

**Comment by dpetka2001** [comment]
You could do it like this
```lua
  {
    "nvim-telescope/telescope.nvim",
    keys = {
      {
        "<leader>fs",
        function()
          require("telescope.builtin").lsp_document_symbols({
            symbols = require("lazyvim.config").get_kind_filter(),
          })
        end,
        desc = "Goto Symbol",
      },
    },
  },
```

**Reply by juridiener** [reply]
Thank you. How can i ignore some symbols like variable, i want just see functions as well as arrow functions in js and methods.

**Reply by dpetka2001** [reply]
You can just define instead of `symbols = require("lazyvim.config").get_kind_filter()`, `symbols = { "function" }` and whatever other symbols you would like to see.

**Reply by juridiener** [reply]
I noticed that in Javascript arrow functions are treated as constant. Is there a way that telescope treats arrow functions as functions?

**Reply by dpetka2001** [reply]
No idea. I believe this comes directly from the LSP server and Telescope just shows the results.

**Reply by juridiener** [reply]
Okay, can i find somewhere a list of the symbols? I searched on the github page from telescope but i could not find any list there.

**Reply by dpetka2001** [reply]
The symbols come from the LSP server, so you would have to search in the github repo of the server you're interested in to see what symbols it supports. I've never done this. What I usually do is just `:Telescope lsp_document_symbols` and see all the symbols and what `kind` (`function`, `constant` etc) they are and then move from there.

**Reply by juridiener** [reply]
okay thanks for your help.
